src/components/forms/pool_actions/MigrateForm/composables/useMigrateMath.ts

```diff
@@ -0,0 +1,119 @@
+import { computed, ref, Ref } from 'vue';
+import { parseUnits } from 'ethers/lib/utils';
+
+import useNumbers from '@/composables/useNumbers';
+import useTokens from '@/composables/useTokens';
+import useUserSettings from '@/composables/useUserSettings';
+
+import { FullPool } from '@/services/balancer/subgraph/types';
+import PoolCalculator from '@/services/pool/calculator/calculator.sevice';
+
+import { bnSum, bnum } from '@/lib/utils';
+import useSlippage from '@/composables/useSlippage';
+
+export type MigrateMathResponse = ReturnType<typeof useMigrateMath>;
+
+export default function useMigrateMath(
+  fromPool: Ref<FullPool>,
+  toPool: Ref<FullPool>
+) {
+  /**
+   * COMPOSABLES
+   */
+  const { tokens, balances, balanceFor, getToken } = useTokens();
+  const { currency } = useUserSettings();
+  const { fNum, toFiat } = useNumbers();
+  const { minusSlippageScaled } = useSlippage();
+
+  /**
+   * SERVICES
+   */
+  const fromPoolCalculator = new PoolCalculator(
+    fromPool,
+    tokens,
+    balances,
+    'exit',
+    ref(false)
+  );
+
+  const toPoolCalculator = new PoolCalculator(
+    toPool,
+    tokens,
+    balances,
+    'join',
+    ref(false)
+  );
+
+  /**
+   * COMPUTED
+   */
+  const bptBalance = computed(() => balanceFor(fromPool.value.address));
+
+  const hasBpt = computed(() => bnum(bptBalance.value).gt(0));
+
+  const tokenCount = computed(() => fromPool.value.tokenAddresses.length);
+
+  const poolDecimals = computed(() => fromPool.value.onchain.decimals);
+
+  const bptBalanceScaled = computed(() =>
+    parseUnits(bptBalance.value, poolDecimals.value).toString()
+  );
+
+  const withdrawalTokens = computed(() =>
+    fromPool.value.tokenAddresses.map(address => getToken(address))
+  );
+
+  const fullAmounts = computed(() => {
```

Comment:
@garethfuller I'm using the name `fullAmounts` and not `proportionalAmounts` because I don't have the same logic as the one in `invest/withdraw`